packages/panzer/CMakeLists.txt

```diff
@@ -28,10 +28,10 @@ SET(${PACKAGE_NAME}_ADD_EXPENSIVE_CUDA_TESTS
   "Add expensive convergence tests to cuda builds.  Not recommended for CMAKE_BUILD_TYPE=DEBUG and CUDA")
 MESSAGE(STATUS "Convergence tests: ${${PACKAGE_NAME}_ADD_EXPENSIVE_CUDA_TESTS}")
 
-TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_FADTYPE
-  PANZER_FADTYPE
-  "Choose the Sacado automatic differentiation scalar type (default is DFad)."
-  "Sacado::Fad::DFad<RealType>" )
```

Comment:
@prwolfe, note that the macro `TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_FADTYPE
  PANZER_FADTYPE... `) sets both the cache vars `${PACKAGE_NAME}_ENABLE_FADTYPE` and `PANZER_FADTYPE` so I think this change may be actually implicitly setting  `PANZER_FADTYPE` to empty '' (i.e. undefined CMake cache vars are implicitly empty).

Reply:
Wait, I got that backwards, you changes this to set `PANZER_FADTYPE` instead of `${PACKAGE_NAME}_ENABLE_FADTYPE`.  But the external cache var that a user sets should be called `${PACKAGE_NAME}_ENABLE_FADTYPE` instead of `PANZER_FADTYPE` so that it sorts properly with the other `Phalanx_` CMake cache vars.

@rppawlo, you might consider changing the name of `${PACKAGE_NAME}_ENABLE_FADTYPE` to just `${PACKAGE_NAME}_FADTYPE` because vars with `ENABLE` in them are typically bool-type vars (with values 'ON', 'OFF', and empty '').  See:

* https://tribits.org/doc/TribitsDevelopersGuide.html#how-to-check-for-and-tweak-tribits-enable-cache-variables

Having a no-bool value for an `_ENABLE_` var is likely to confuse/surprise people.

Reply:
But there is no case in the source code that will work if the value is OFF. So why do I need the _ENABLE_ option? I eliminated it because it was not functional. When it is set to OFF the result is a compile failure as the config file does not define PANZER_FADTYPE.